dotnet/src/Agents/Yaml/KernelAgentFactoryYamlExtensions.cs

```diff
@@ -13,17 +13,17 @@ public static class KernelAgentFactoryYamlExtensions
     /// <summary>
     /// Create a <see cref="KernelAgent"/> from the given YAML text.
     /// </summary>
-    /// <param name="kernelAgentFactory">Kernel agent factory which will be used to create the agent.</param>
-    /// <param name="kernel">Kernel instance.</param>
+    /// <param name="kernelAgentFactory">Kernel agent factory which will be used to create the agent</param>
     /// <param name="text">Text string containing the YAML representation of a kernel agent.</param>
-    /// <param name="cancellationToken">Optional cancellation token.</param>
-    public static async Task<KernelAgent?> CreateAgentFromYamlAsync(this KernelAgentFactory kernelAgentFactory, Kernel kernel, string text, CancellationToken cancellationToken = default)
+    /// <param name="kernel">Kernel instance</param>
+    /// <param name="cancellationToken">Optional cancellation token</param>
+    public static async Task<KernelAgent?> CreateAgentFromYamlAsync(this KernelAgentFactory kernelAgentFactory, string text, Kernel? kernel = null, CancellationToken cancellationToken = default)
     {
         var agentDefinition = AgentDefinitionYaml.FromYaml(text);
         agentDefinition.Type = agentDefinition.Type ?? (kernelAgentFactory.Types.Count > 0 ? kernelAgentFactory.Types[0] : null);
 
         return await kernelAgentFactory.CreateAsync(
-            kernel,
+            kernel ?? new Kernel(),
```

Comment:
Having an empty kernel here can bring us some problems, if this is required for the underlying `Create` we might consider requiring it from the `CreateAgentFrom...`.

```suggestion
            kernel ?? new Kernel(),
```

Reply:
This new pattern is intentional, the `Kernel` should be optional. This is part of the new Agent first initiative.